grammar.js

```diff
@@ -560,11 +580,33 @@ module.exports = grammar({
 
     finally_clause: $ => prec.right(seq('finally', $.expression)),
 
+    binding: $ => prec.dynamic(PREC.binding, seq(
+        field('name', $.identifier),
+        optional(seq(':', field('type', $._param_type))),
+    )),
+
+    bindings: $ => seq(
+        '(',
+        commaSep($.binding),
+        ')',
+    ),
+
+    lambda_expression: $ => prec.right(PREC.lambda, seq(
+        choice(
+            $.bindings,
+            $.identifier,
+        ),
+        '=>',
+        $.expression,
+    )),
+
     case_block: $ => choice(
       prec(-1, seq('{', '}')),
-      seq('{', repeat1($.case_clause), '}')
+      $._case_clauses,
     ),
 
+    _case_clauses: $ => seq('{', repeat1($.case_clause), '}'),
```

Comment:
Just curious - what was the purpose of extracting this rule?

Reply:
Hmm good catch, it's not needed. I remember adding it because it's used in the `call_expression`, to dissalow `f{}`, but that's a valid expression.